### PR TITLE
refactored patient scheduled visits and added resource file

### PR DESCRIPTION
**Pull request by CynthiaKamau**
## Requirements

- [ ] This PR has a title that briefly describes the work done, including the ticket number if there is a ticket.
- [ ] My work conforms to the [**OpenMRS 3.0 Styleguide**](https://om.rs/styleguide).
- [ ] I checked for feature overlap with [**existing widgets**](https://om.rs/directory).

## Summary
Refactored patient scheduled visits and added resource file

## Screenshots

*None.*

## Related Issue

*None.*

## Other

*None.*

### CODE REVIEW

**Comment by CynthiaKamau** [comment]
Kindly review @ZacButko @donaldkibet @denniskigen

File: packages/esm-outpatient-app/src/patient-search/patient-scheduled-visits.component.tsx [hunk 1]
```diff
@@ -70,101 +181,9 @@ const PatientScheduledVisits: React.FC<PatientSearchProps> = ({ toggleSearchType
...
-                    />
-                  </ContentSwitcher>
-                ) : null}
-              </div>
-            </RadioTile>
-          ))}
-        </TileGroup>
-      </div>
+      <PatientRecentScheduledVisits recentVisits={recentVisits} isLoading={isLoading} />
+      <PatientFutureScheduledVisits futureVisits={futureVisits} isLoading={loading} />
```
Comment (donaldkibet) [inline packages/esm-outpatient-app/src/patient-search/patient-scheduled-visits.component.tsx:185]: Basically, these two components are duplicating code, achieving almost identical results, you can refactor it to something like 


```
<ScheduledVisits visitType='' visits={[]}/>
```

where you pass the visit type as the title for that visit either `recent` or `future` and visits array prop which can either be `recentVisits` or `futureVisits`

**Review by donaldkibet — APPROVED** [review]
Thanks, @CynthiaKamau! few suggestions

File: packages/esm-outpatient-app/src/patient-search/patient-scheduled-visits.component.tsx [hunk 2]
```diff
@@ -1,13 +1,22 @@
...
+  RadioTile,
+  TileGroup,
+  DataTableSkeleton,
+} from 'carbon-components-react';
 import ArrowLeft24 from '@carbon/icons-react/es/arrow--left/24';
 import { formatDatetime, useLayoutType, parseDate } from '@openmrs/esm-framework';
 import { SearchTypes } from '../types';
 import styles from './patient-scheduled-visits.scss';
-
+import { useRecentScheduledVisits, useFutureScheduledVisits } from './patient-scheduled-visits.resource';
```
[the remark this answers is not in the document]
Comment (CynthiaKamau) [reply packages/esm-outpatient-app/src/patient-search/patient-scheduled-visits.component.tsx:16]: There two different hooks in the resource file

File: packages/esm-outpatient-app/src/patient-search/patient-scheduled-visits.component.tsx [hunk 3]
```diff
@@ -16,46 +25,84 @@ enum priority {
...
+
+const ScheduledVisits: React.FC<{ visits; isLoading; visitType }> = ({ visits, isLoading, visitType }) => {
   const { t } = useTranslation();
-  const isTablet = useLayoutType() === 'tablet';
-  const [showRecentPriority, setShowRecentPriority] = useState(false);
-  const [futurePrioritySwitcherValue, setFuturePrioritySwitcherValue] = useState(0);
-  const [futureVisitsIndex, setFutureVisitsIndex] = useState(0);
+  const [prioritySwitcherValue, setSwitcherValue] = useState(0);
+  const [visitsIndex, setVisitsIndex] = useState(0);
+  const [show_priority, setShowPriority] = useState(false);
```
Comment (donaldkibet) [inline packages/esm-outpatient-app/src/patient-search/patient-scheduled-visits.component.tsx:37]:
```suggestion
  const [hasPriority, setHasPriority] = useState(false);
```
The project uses `camelCase` for consistency purposes we should stick to it

File: packages/esm-outpatient-app/src/patient-search/patient-scheduled-visits.component.tsx [hunk 4]
```diff
@@ -16,46 +25,84 @@ enum priority {
...
+    return <DataTableSkeleton role="progressbar" />;
+  }
 
-  const [show_future_priority, setShowFuturePriority] = useState(false);
-  const [recentPrioritySwitcherValue, setRecentPrioritySwitcherValue] = useState(0);
-  const [recentVisitsIndex, setRecentVisitsIndex] = useState(0);
+  if (visits) {
+    return (
+      <div>
+        {visits.length >= 1 ? (
```
Comment (donaldkibet) [inline packages/esm-outpatient-app/src/patient-search/patient-scheduled-visits.component.tsx:46]:
```suggestion
        {visits.length > 0 ? (
```
nit

File: packages/esm-outpatient-app/src/patient-search/patient-scheduled-visits.component.tsx [hunk 5]
```diff
@@ -16,46 +25,84 @@ enum priority {
...
+          <div className={styles.row}>
+            <p className={styles.heading}>{t('futureScheduledVisits', { count: visits.length })} </p>
```
Comment (donaldkibet) [inline packages/esm-outpatient-app/src/patient-search/patient-scheduled-visits.component.tsx:48]:
```suggestion
            <p className={styles.heading}>{t('visitsCount', { count: visits.length })} </p>
```

Comment (CynthiaKamau) [reply packages/esm-outpatient-app/src/patient-search/patient-scheduled-visits.component.tsx]: Done

**Pull request merged**